docs/cnn.md

```diff
@@ -74,23 +74,285 @@ conv = ConvModule(
+    ```
+
+- kaiming_init
+
+  Initialize module paramters with the valuse according to the method
+  described in [Delving deep into rectifiers: Surpassing human-level
+  performance on ImageNet classification - He, K. et al. (2015)](https://www.cv-foundation.org/openaccess/content_iccv_2015/papers/He_Delving_Deep_into_ICCV_2015_paper.pdf)
+
+    ```python
+    >>> import torch.nn as nn
+    >>> from mmcv.cnn.utils.weight_init import kaiming_init
+    >>> conv1 = nn.Conv2d(3, 3, 1)
+    >>> # kaiming_init(module, a=0, mode='fan_out', nonlinearity='relu', bias=0, distribution='normal')
+    >>> kaiming_init(conv1)
+    ```
+
+- caffe2_xavier_init
+  Corresponds to `kaiming_uniform_` in PyTorch.
+
+    ```python
+    >>> import torch.nn as nn
+    >>> from mmcv.cnn.utils.weight_init import caffe2_xavier_init
+    >>> conv1 = nn.Conv2d(3, 3, 1)
+    >>> # caffe2_xavier_init(module, bias=0)
+    >>> caffe2_xavier_init(conv1)
+    ```
+
+- bias_init_with_prob
+
+  Initialize conv/fc bias value according to given probability proposed in [Focal Loss for Dense Object Detection](https://arxiv.org/pdf/1708.02002.pdf).
+
+    ```python
+    >>> from mmcv.cnn.utils.weight_init import bias_init_with_prob
+    >>> # bias_init_with_prob is proposed in Focal Loss
+    >>> bias = bias_init_with_prob(0.01)
+    >>> bias
+    -4.59511985013459
+    ```
+
+#### **Initialization of model**
+
+On the basis of the initialization methods, we define the corresponding initialization classes and register them to `INITIALIZERS`, so we can
+use the configuration to initialize the model.
+
+We provide the following initialization classes.
+
+- BaseInit
+- BaseInit
+- XavierInit
+- NormalInit
+- UniformInit
+- KaimingInit
+- Caffe2XavierInit
+- PretrainedInit
+
+Before we go deeper into the usage of `initialize`, briefly introducing the
+design principle of it is helpful.
+
+- If we don't define `layer` key or `override` key, it will not initialize anything.
+- If we define `override` but don't define `layer`, it will initialize parameters with the attribute name in `override`.
+- If we only define `layer`, it just initialize the layer in `layer` key.
+- If we define `override` and `layer`, `override` has higher priority and will override initialization mechanism.
+
+Now, it is time to introduce the usage of `initialize` in detail.
+
+- Initialize whole module with the same configuration
+
+  Define `layer` for initializing layer with same configuration.
+
+    ```python
+    import torch.nn as nn
+    from mmcv.cnn.utils.weight_init import initialize
+
+    class FooNet(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.feat = nn.Conv1d(3, 1, 3)
+            self.reg = nn.Conv2d(3, 3, 3)
+            self.cls = nn.Linear(1, 2)
+
+    model = FooNet()
+    init_cfg = dict(type='Constant', layer=['Conv1d', 'Conv2d', 'Linear'], val=1)
+    # initialize whole module with same configuration
+    initialize(model, init_cfg)
+    ```
```

Comment:
I strongly recommend that write this example after initialization classes list, and introduce keys in `init_cfg` after the example. Others functions of init_cfg can also be introduced by examples.

Reply:
got it